property/index.js

```diff
@@ -115,54 +118,128 @@ module.exports = yeoman.generators.Base.extend({
       {
          name: 'defaultValue',
          message: 'Default value[leave blank for none]:',
-         default: null
+         default: null,
+         when: function(answers) {
+          return answers.type !== null && 
+            answers.type !== 'buffer'&&
+            answers.type !== 'any'&&
+            typeChoices.indexOf(answers.type) !== -1;
+        }
       }
     ];
+
     this.prompt(prompts, function(answers) {
+      debug('answers: %j', answers);
       this.name = answers.name || this.name;
       if (answers.type === 'array') {
         var itemType =  answers.customItemType || answers.itemType;
         this.type = itemType ? [itemType] : 'array';
       } else {
         this.type = answers.customType || answers.type;
       }
-      this.required = answers.required;
-      this.defaultValue = answers.defaultValue;
+
+      this.propDefinition = {
+        name: this.name,
+        type: this.type
+      };
+      if (answers.required) {
+        this.propDefinition.required = true;
```

Comment:
Nitpick: Is there any particular reason why not set `required` to `answers.required` directly? If you want to coerce to a boolean, then you can use `!!answers.required` or `Boolean(answers.required)`, but I would expect inquirer to already provide a boolean value.


```js
this.propDefinition = {
  name: this.name,
  type: this.type,
  required: this.propDefinition.required,
};
```